internal/venice-common/src/main/java/com/linkedin/venice/blobtransfer/DaVinciBlobFinder.java

```diff
@@ -22,32 +27,48 @@
 /**
  * DvcBlobFinder discovers live DaVinci peer nodes to facilitate blob transfers necessary for bootstrapping the database
  */
-public class DvcBlobFinder implements BlobFinder {
-  private static final Logger LOGGER = LogManager.getLogger(DvcBlobFinder.class);
+public class DaVinciBlobFinder implements BlobFinder {
+  private static final Logger LOGGER = LogManager.getLogger(DaVinciBlobFinder.class);
   private static final String TYPE_BLOB_DISCOVERY = "blob_discovery";
   private static final String ERROR_DISCOVERY_MESSAGE =
       "Error finding DVC peers for blob transfer in store: %s, version: %d, partition: %d";
-  private final TransportClient transportClient;
+  private final AvroGenericStoreClientImpl storeClient;
 
-  public DvcBlobFinder(TransportClient transportClient) {
-    this.transportClient = transportClient;
+  public DaVinciBlobFinder(AvroGenericStoreClientImpl storeClient) {
+    this.storeClient = storeClient;
   }
 
   @Override
   public BlobPeersDiscoveryResponse discoverBlobPeers(String storeName, int version, int partition) {
-    String uri = buildUriForBlobDiscovery(storeName, version, partition);
+    String requestPath = buildUriForBlobDiscovery(storeName, version, partition);
+    byte[] response = executeRequest(requestPath);
 
-    CompletableFuture<BlobPeersDiscoveryResponse> futureResponse = transportClient.get(uri).thenApply(response -> {
-      byte[] responseBody = response.getBody();
-      ObjectMapper mapper = ObjectMapperFactory.getInstance();
-      try {
-        return mapper.readValue(responseBody, BlobPeersDiscoveryResponse.class);
-      } catch (IOException e) {
-        return handleError(ERROR_DISCOVERY_MESSAGE, storeName, version, partition, e);
-      }
-    }).exceptionally(throwable -> handleError(ERROR_DISCOVERY_MESSAGE, storeName, version, partition, throwable));
+    ObjectMapper mapper = ObjectMapperFactory.getInstance();
+    BlobPeersDiscoveryResponse discoveryResponse;
+    try {
+      discoveryResponse = mapper.readValue(response, BlobPeersDiscoveryResponse.class);
+      return discoveryResponse;
+    } catch (IOException e) {
+      return handleError(ERROR_DISCOVERY_MESSAGE, storeName, version, partition, e);
+    }
+  }
+
+  private byte[] executeRequest(String requestPath) {
+    byte[] response;
+    try {
+      response = RetryUtils.executeWithMaxAttempt(
+          () -> ((CompletableFuture<byte[]>) storeClient.getRaw(requestPath)).get(),
+          3,
+          Duration.ofSeconds(5),
+          Collections.singletonList(ExecutionException.class));
+    } catch (Exception e) {
+      throw new VeniceException("Failed to fetch blob peers from path " + requestPath, e);
```

Comment:
This would throw exception at line 44, then it need to be handled by the caller of `discoverBlobPeers`. This is different than the previous behavior where all exceptions are supposed to be caught and return a `BlobPeersDiscoveryResponse` has `Error` set.

Regardless whether we want to explicitly throw exceptions or catch them and return a POJO, I think we should make the behavior consistent in this method

Reply:
sounds good, I have changed some of the logic in my latest commit so please let me know if it is what you meant